isis/tests/Area3DTests.cpp

```diff
@@ -0,0 +1,281 @@
+  Isis::Area3D area3(area1.intersect(area2));
+  EXPECT_FALSE(area3.isValid());
+}
+
+TEST(Area3D, Setters)
+{
+  Isis::Area3D area1;
+  Isis::Displacement start(0, Isis::Displacement::Meters);
+  Isis::Displacement end(10, Isis::Displacement::Meters);
+  Isis::Displacement move(3, Isis::Displacement::Meters);
+  Isis::Distance dim(5, Isis::Distance::Meters);
+
+  area1.setStartX(start);
+  area1.setStartY(start);
+  area1.setStartZ(start);
+  area1.setEndX(end);
+  area1.setEndY(end);
+  area1.setEndZ(end);
+  EXPECT_DOUBLE_EQ(area1.getEndX().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getEndY().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getEndZ().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getStartX().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+
+  area1.setXDimension(start, dim);
+  area1.setYDimension(start, dim);
+  area1.setZDimension(start, dim);
+  EXPECT_DOUBLE_EQ(area1.getEndX().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getEndY().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getEndZ().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getStartX().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+
+  area1.moveEndX(move);
+  area1.moveEndY(move);
+  area1.moveEndZ(move);
+  EXPECT_DOUBLE_EQ(area1.getEndX().meters(), 3);
+  EXPECT_DOUBLE_EQ(area1.getEndY().meters(), 3);
+  EXPECT_DOUBLE_EQ(area1.getEndZ().meters(), 3);
+  EXPECT_DOUBLE_EQ(area1.getStartX().meters(), -2);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), -2);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), -2);
+
+  move.setMeters(0);
+  area1.moveStartX(move);
+  area1.moveStartY(move);
+  area1.moveStartZ(move);
+  EXPECT_DOUBLE_EQ(area1.getEndX().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getEndY().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getEndZ().meters(), 5);
+  EXPECT_DOUBLE_EQ(area1.getStartX().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+
+  dim.setMeters(10);
+  area1.setDepth(dim);
+  area1.setHeight(dim);
+  area1.setWidth(dim);
+  EXPECT_DOUBLE_EQ(area1.getEndX().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getEndY().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getEndZ().meters(), 10);
+  EXPECT_DOUBLE_EQ(area1.getStartX().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+  EXPECT_DOUBLE_EQ(area1.getStartY().meters(), 0);
+ }
+
+TEST(Area3D, Operators)
+{
+  Isis::Displacement start(5, Isis::Displacement::Meters);
+  Isis::Displacement end(10, Isis::Displacement::Meters);
+  Isis::Distance dim(5, Isis::Distance::Meters);
+  Isis::Area3D area1(start, start, start, end, end, end);
+  Isis::Area3D area2(start, start, start, dim, dim, dim);
+  EXPECT_TRUE(area1 == area2);
+  EXPECT_FALSE(area1 != area2);
+
+  start.setMeters(1);
+  area1.setStartX(start);
+  EXPECT_FALSE(area1 == area2);
+  EXPECT_TRUE(area1 != area2);
+}
```

Comment:
Should there be a test for the = operator (assigning an area to another)?

Reply:
Thanks, I added the test for the = operator